src/huggingface_hub/hf_api.py

```diff
@@ -240,6 +315,9 @@ def login(self, username: str, password: str) -> str:
         r = requests.post(path, json={"username": username, "password": password})
         r.raise_for_status()
         d = r.json()
+
+        cased_username = self.whoami(d["token"])["name"]
```

Comment:
No strong opinion.

is this the same as `username.lower()`? Maybe we could save an API call here, but maybe this is safer to be consistent with `erase_from_credential_store`

Reply:
Absolutely! I went ahead the "correctly" cased way - but any casing works as long as it's identical across the helper methods. Thanks!